### ISSUE TITLE
Forest area as proportion of total land area SDG - should be estimated bases on the latest available NDP after 2020

### ISSUE THREAD

**Issue by pengchenglai** [openforis/fra-platform#3203]
@minotogna 

**Describe the bug**
 **Forest area as proportion of total land area** SDG indicator uses the linear trend even though there is a new NDP after 2020.

**To Reproduce**
See Australia(screenshots below)

**Expected behavior**
If the latest data of NDP after 2020 is available, the **Forest area as proportion of total land area** should be estimated based on the latest data of NDP **not the 2020 data**

**Screenshots**
![image](https://github.com/openforis/fra-platform/assets/99216009/387418d5-57f5-4e8a-8883-4b2bfac8c6d9)
![image](https://github.com/openforis/fra-platform/assets/99216009/524c2d7f-9b89-4a9e-88ed-3a604cfe3c97)

**Comment by sorja** [comment]
@pengchenglai After working with this I think we must take into account several cases
Reference PR: https://github.com/openforis/fra-platform/pull/3427
Could you be so kind to verify I did not miss any?

---

- Table SDG 15.1.1

---

- We are solely discussing the years 2021, 2022, 2023 and 2024.
- These fields are the only editable fields
- These fields are interpolated from existing years


---

- Case 1: No data
    - Show null
- Case 2: Data only for either year 2020 or 2025
    - Show null
- Case 3: Data for both 2020 and 2025 but not between
  - Interpolate using 2020 and 2025 years 2021...2024

---


- Case 4: Data for 2020 and 2021 and 2025
  - Extrapolate using 2020 and 2021?
Case 5: Data for 2020 and 2021
  - Extrapolate using 2020 and 2021
- Case 6: Data for 2021...2023
  - Extrapolate 2024 using values from year 2022 and 2023
- Case 7: Data for 2021 and 2023
  - Interpolate 2022 using 2021 and 2023
  - Extrapolate 2024 using 2021 and 2023

---

Let me know if you can think of any other cases and if any of the above logic should be changed.


In short:
- If no NDPs exist for years 2021...2024 we use 2020 and 2025.
- If either or both of the values don't exist, then return empty

Questions:
- If single value exists exists, is it preferred to use 2020 or 2025 to interpolate / extrapolate?
- If two value or more values exist, should we use the NDPs always to interpolate / extrapolate and ignore values in 2020 and 2025?

Thanks

**Comment by pengchenglai** [comment]
@sorja 

Thank you. In reality, this scenario usually does not occur because the data for 2025 would be repeated from the data for 2020 or any existing data reported in previous reports if NC does not have it for 2025. : _If single value exists exists, is it preferred to use 2020 or 2025 to interpolate / extrapolate?_

**Comment by pengchenglai** [comment]
@sorja 

We could have a quick talk later today or tomorrow to resolve this issue which might make it easy to clarify the cases if possible

**Comment by minotogna** [comment]
@pengchenglai this issue is not 100% clear:
here is an example of the formula for year 2021.

```
extentOfForest.forestArea['2021'] ? 
                extentOfForest.forestArea['2021'] / extentOfForest.totalLandArea['2021'] * 100 : 
                (
                  (extentOfForest.forestArea['2020'] / extentOfForest.totalLandArea['2020'] * 100 > 0 && extentOfForest.forestArea['2025'] / extentOfForest.totalLandArea['2025'] * 100) > 0 ? 
                    (extentOfForest.forestArea['2020'] / extentOfForest.totalLandArea['2020'] * 100 + ((extentOfForest.forestArea['2025'] / extentOfForest.totalLandArea['2025'] * 100 - extentOfForest.forestArea['2020'] / extentOfForest.totalLandArea['2020'] * 100) / 5 * 1)) 
                    : null
                )
```

how should we modify this ?
thank you

**Comment by pengchenglai** [comment]
@minotogna 

True, the code works for 2021. However, if the data is for 2023, then **we need to add one more step to check if there are any NDPs available between 2020 and 2025**.

In the case of Australia, there is no NDP for 2023, but there are NDPs for 2021 and 2022. Therefore, data from 2022 and 2025 should be used for linear estimation for the Forest area as proportion of total land area.

**Comment by minotogna** [comment]
> @minotogna 
> 
> 
> 
> True, the code works for 2021. However, if the data is for 2023, then **we need to add one more step to check if there are any NDPs available between 2020 and 2025**.
> 
> 
> 
> In the case of Australia, there is no NDP for 2023, but there are NDPs for 2021 and 2022. Therefore, data from 2022 and 2025 should be used for linear estimation for the Forest area as proportion of total land area.
> 
> 
> 
> 

@pengchenglai @sorja can we have a chat tomorrow morning 10am Rome time ?

**Comment by pengchenglai** [comment]
If we now use the data from 2020 and 2025 to estimate the data for 2023, the value will be lower than that of 2022. However, we know it should be the same as the value for 2022, which is 17.36.

![image](https://github.com/openforis/fra-platform/assets/99216009/f6ee2873-a468-42ad-ae90-1d24d8c41968)

**Comment by pengchenglai** [comment]
@minotogna 

For me it is fine
